components/domains/externalDomainTransferModal.tsx

```diff
@@ -0,0 +1,168 @@
+  const [targetAddress, setTargetAddress] = useState<string>("");
+  const { addTransaction } = useNotificationManager();
+  const [addressInput, setAddressInput] = useState<string>("");
+  const { starknetIdNavigator } = useContext(StarknetIdJsContext);
+  const [isTxSent, setIsTxSent] = useState(false);
+  const [callData, setCallData] = useState<Call[]>([]);
+
+  const { writeAsync: transfer_name, data: transferData } = useContractWrite({
+    calls: callData,
+  });
+
+  useEffect(() => {
+    setCallData([
+      {
+        contractAddress: resolverContract,
+        entrypoint: "transfer_name",
+        calldata: [domainEncoded, targetAddress],
+      },
+    ]);
+  }, [domainEncoded, targetAddress, resolverContract]);
+
+  useEffect(() => {
+    if (isHexString(addressInput)) {
+      setTargetAddress(addressInput);
+    } else if (utils.isStarkDomain(addressInput)) {
+      starknetIdNavigator?.getAddressFromStarkName(addressInput).then((res) => {
+        if (!res || res === "0x0") setTargetAddress("");
+        else setTargetAddress(res);
+      });
+    } else {
+      setTargetAddress("");
+    }
+  }, [addressInput, starknetIdNavigator]);
+
+  useEffect(() => {
+    if (!transferData?.transaction_hash) return;
+    addTransaction({
+      timestamp: Date.now(),
+      subtext: `${domain} transfered`,
+      type: NotificationType.TRANSACTION,
+      data: {
+        type: TransactionType.TRANSFER_IDENTITY,
+        hash: transferData.transaction_hash,
+        status: "pending",
+      },
+    });
+    setIsTxSent(true);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [transferData]); // We want to call this only once when the transaction is sent
+
+  function transferName(): void {
+    transfer_name();
+  }
+
+  function changeAddress(value: string): void {
+    setAddressInput(value);
+  }
+
+  function closeModal(): void {
+    setIsTxSent(false);
+    handleClose();
+  }
+
+  return (
+    <Modal
+      disableAutoFocus
+      open={isModalOpen}
+      onClose={closeModal}
+      aria-labelledby="modal-modal-title"
+      aria-describedby="modal-modal-description"
+    >
+      {isTxSent ? (
+        <ConfirmationTx
+          closeModal={closeModal}
+          txHash={transferData?.transaction_hash}
+        />
+      ) : (
+        <div className={styles.menu}>
+          <button className={styles.menu_close} onClick={closeModal}>
+            <svg viewBox="0 0 24 24">
+              <path
+                strokeLinecap="round"
+                strokeLinejoin="round"
+                strokeWidth="2"
+                d="M6 18L18 6M6 6l12 12"
+              ></path>
+            </svg>
+          </button>
+          <h2 className={styles.menu_title}>
+            Transfer your domain to a different wallet
+          </h2>
+          {domainKind ? (
+            <p className="mt-5 text-center">
+              You can only transfer your subdomain to a {domainKind} wallet.
```

Comment:
Doesn't work for all of our cases (Xplorer wallet ? Sol wallet ?)

<img width="206" alt="Capture d’écran 2024-04-26 à 11 10 12" src="https://github.com/starknet-id/app.starknet.id/assets/78437165/37bba8c5-65f1-4430-a9d6-9e362e2f8cfe">

Reply:
We can only transfer xplorer and braavos subdomains, but I'm not sure I understood your comment

Reply:
Ah you mean the return ? Actually the value is the result of the function `getResolverName`

Reply:
I updated this to a new function called `getResolverCondition(domainKind: DomainKind)` in the transfer modal which returns the text to show, because it's really particular to this modal and shouldn't be needed elsewhere.